PackageItem schema test

Jeshan7 opened the pull request:
## Why?
  - Coverage was low

## This change addresses the need by:
     - Updated the test cases in schema/package_item_test.exs
     - Wrote new test cases in schema/package_item_test.exs
        - Test cases for foreign_keys(product_id, packege_id, line_item_id)
        - Test cases for update_changeset/2

    [delivers #163248592]

## Checklist
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ ] I have read [CONTRIBUTING.md][contributing].
- [ ] My code follows the [style guidelines][contributing] of this project.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] My code does not generate any (new) [`credo`][credo] and compile-time warnings.
- [ ] I have updated the documentation wherever necessary.
- [ ] I have added tests to cover my changes.

[contributing]: https://github.com/aviabird/snitch/CONTRIBUTING.md
[credo]: https://github.com/rrrene/credo

Jeshan7 committed:
[#163248592]
   - Updated the test cases in schema/package_item_test.exs
   - Wrote new test cases in schema/package_item_test.exs

SagarKarwande commented on apps/snitch_core/test/data/schema/package_item_test.exs:
```diff
@@ -19,8 +20,12 @@ defmodule Snitch.Data.Schema.PackageItemTest do
+
   describe "create_changeset/2" do
-    test "with valid params, and backorder is computed correctly" do
+    test "suuceesfully with valid params, and backorder is computed correctly" do
```
Typo `suuceesfully`

SagarKarwande commented on apps/snitch_core/test/data/schema/package_item_test.exs:
```diff
@@ -91,5 +123,79 @@ defmodule Snitch.Data.Schema.PackageItemTest do
+    end
+  end
+
+  defp new_package(context) do
```
This method setup the context with things related to package.

Can we rename this function with proper name?
**OR**
We can setup thing individually in a particular order as follows:
```elixir
setup :product
setup :line_item
setup :package
```
@arjun289 Any thoughts on this?

arjun289 replied:
Yes it can be done this way

SagarKarwande commented on apps/snitch_core/test/data/schema/package_item_test.exs:
```diff
@@ -91,5 +123,79 @@ defmodule Snitch.Data.Schema.PackageItemTest do
+      refute cs.valid?
+
+      package_item = %{package_item | tax: bad_money, shipping_tax: bad_money}
+      cs = PackageItem.update_changeset(%PackageItem{}, Map.from_struct(package_item))
```
We should use already inserted Package item for update.

SagarKarwande commented on apps/snitch_core/test/data/schema/package_item_test.exs:
```diff
@@ -91,5 +123,79 @@ defmodule Snitch.Data.Schema.PackageItemTest do
+      refute cs.valid?
+
+      package_item = %{package_item | quantity: -3, delta: -2}
+      cs = PackageItem.update_changeset(%PackageItem{}, Map.from_struct(package_item))
```
Use already inserted package item for update

SagarKarwande reviewed (changes requested):
Format the files with Elixir formatter

arjun289 commented on apps/snitch_core/test/data/schema/package_item_test.exs:
```diff
@@ -41,7 +46,34 @@ defmodule Snitch.Data.Schema.PackageItemTest do
     end
 
-    test "with invalid quantity, delta" do
+    test "fails for non-existent line_item_id", context do
```
Group together the tests for missing foreign keys in one test.

Jeshan7 replied:
By grouping them together it checks only for one foreign key constraint and ignores the other

Jeshan7 committed:
Prior to this change,
 - Two separate functions were present to create package and package_item
 - Update_changset/2 was adding params to empty PackageItem struct
This change
 - Line_item , product and package has its own function
 - update_changset/2 now adds params to the struct which is already
   present in database.

Jeshan7 committed:
Merge branch 'develop' into package_item_schema_test
